tests/xua_sim_tests/test_i2s_loopback/src/main.xc

```diff
@@ -211,7 +212,44 @@ clock clk_audio_lrclk_gen = on tile[AUDIO_IO_TILE] : XS1_CLKBLK_5;
 void slave_mode_clk_setup(const unsigned samFreq, const unsigned chans_per_frame);
 #endif
 
-#endif
+
+void mclk_checker(void)
+{
+  if(AUDIO_IO_TILE == XUD_TILE)
+  {
+    // printstrln("FAAILLE");
+
+    int x;
+    asm("ldw %0, dp[clk_audio_mclk]":"=r"(x));
+    asm("setclk res[%0], %1"::"r"(p_for_mclk_count), "r"(x));
+
+    unsigned c0, c1;
+    // startup value of port timer
+    delay_microseconds(5); // Wait for I2S to start the clock
+    p_for_mclk_count :> void @ c0;
+    delay_microseconds(10); // Wait for port timer to increment a bit more. Should be just over 25 ticks per microsecond, so 250
+
+    // Input will fail if not clocked so have timeout
+    timer t;
+    int timeout;
+    t :> timeout;
+
+    select{
+      case t when timerafter(timeout + XS1_TIMER_MHZ) :> int _: // 1 us later
```

Comment:
this is only test code, but timeout+XS1_TIMER_MHZ reads pretty odd. 

I think 

timeout += XS1_TIME_MHZ 
then timerafter(timeout)

This is easier to understand. Using a MHZ value as a delay is a bit odd though.